linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -540,7 +543,26 @@ private[optimizer] abstract class OptimizerCore(
         ArrayLength(transformExpr(array))
 
       case ArraySelect(array, index) =>
-        ArraySelect(transformExpr(array), transformExpr(index))(tree.tpe)
+        val newArray = transformExpr(array)
+
+        val newElemType = newArray.tpe match {
+          case NullType =>
```

Comment:
```suggestion
          case NothingType | NullType =>
```

Reply:
:facepalm: in fact, this made me realize we can always type the `ArraySelect` as `NothingType` in this branch.